### ISSUE TITLE
Don't report shebang-related errors on comments that are not shebangs

### ISSUE THREAD

**Issue by palapapa** [koalaman/shellcheck#3103]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1008, SC2096 etc.
- My shellcheck version (`shellcheck --version` or "online"): 0.9.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
# /etc/profile: login shell setup
#
# That this file is used by any Bourne-shell derivative to setup the
# environment for login shells.
#

# Load environment settings from profile.env, which is created by
# env-update from the files in /etc/env.d
if [ -e /etc/profile.env ] ; then
	. /etc/profile.env
fi

# You should override these in your ~/.bashrc (or equivalent) for per-user
# settings.  For system defaults, you can add a new file in /etc/profile.d/.
export EDITOR=${EDITOR:-/bin/nano}
export PAGER=${PAGER:-/usr/bin/less}

unset ROOTPATH

# process *.sh files in /etc/profile.d
for sh in /etc/profile.d/*.sh ; do
	[ -r "$sh" ] && . "$sh"
done
unset sh

if [ -n "${BASH_VERSION-}" ] ; then
	# Newer bash ebuilds include /etc/bash/bashrc which will setup PS1
	# including color.  We leave out color here because not all
	# terminals support it.
	if [ -f /etc/bash/bashrc ] ; then
		# Bash login shells run only /etc/profile
		# Bash non-login shells run only /etc/bash/bashrc
		# Since we want to run /etc/bash/bashrc regardless, we source it
		# from here.  It is unfortunate that there is no way to do
		# this *after* the user's .bash_profile runs (without putting
		# it in the user's dot-files), but it shouldn't make any
		# difference.
		. /etc/bash/bashrc
	else
		PS1='\u@\h \w \$ '
	fi
else
	# Setup a bland default prompt.  Since this prompt should be useable
	# on color and non-color terminals, as well as shells that don't
	# understand sequences such as \h, don't put anything special in it.
	PS1="${USER:-$(whoami 2>/dev/null)}@$(uname -n 2>/dev/null) \$ "
fi
```

#### Here's what shellcheck currently says:

Gives SC2096

#### Here's what I wanted or expected to see:

All shebang-related checks are triggered even if the first line is just a comment and not a shebang. Disbling the check isn't an option because doing so may suppress the error when there is actually an error.

**Comment by wileyhy** [comment]
Hm.. It's common for beginners to write malformed hashbangs, you know. How
should shellcheck know when a comment is intended to not be a hashbang?
Perhaps it could be something simple like, '###" as the first three
characters of the first line?

  1 # shellcheck disable=SC1113 shell=bash
  2 # /etc/profile: login shell setup
  3 # Shellcheck disable=SC1008,SC2096
  4 #
  5 # That this file is used by any Bourne-shell derivative to setup the
  6 # environment for login shells.
  7 #

Putting the directives on or near the first line does resolve the issue,
but it's sort of a lot to ask people to put any text besides a filename or
a hashbang on the first line, if you ask me.

It might be good form if shellcheck also accepted the " disable=SC1113
shell=bash" directives from line 2 on. Just throwing out some ideas.

Cheers,
Wiley

**Comment by palapapa** [comment]
It would be best if only the errors about malformed shebang are shown. That way I can easily disable them. Right now, even though the first line isn't a shebang, SC still internally thinks it is and shows errors as if it were a shebang. If I disable them, they will also be disabled even when the shebang is actually a correct shebang and will potentially hide real errors.
